optuna_dashboard/ts/components/TrialList.tsx

```diff
@@ -450,204 +490,244 @@ export const TrialList: FC<{ studyDetail: StudyDetail | null }> = ({
 
   return (
     <Box
-      component="div"
-      sx={{ display: "flex", flexDirection: "row", width: "100%" }}
+      sx={{
+        display: "flex",
+        flexDirection: "column",
+        width: "100%",
+        height: "100%",
+      }}
     >
+      {llmEnabled && (
+        <Box
+          component="div"
+          sx={{
+            height: theme.spacing(8),
+            p: theme.spacing(1),
+            display: "flex",
+            flexDirection: "row",
+            alignItems: "center",
+          }}
+        >
+          <TextField
+            id="trial-filter-query"
+            variant="outlined"
+            placeholder="Filter trials. Enter query and press Enter."
+            fullWidth
+            size="small"
+            sx={{
+              maxWidth: "400px",
+            }}
+            value={filterInput}
+            onChange={(e) => setFilterInput(e.target.value)}
+            onKeyDown={(e) => {
+              if (e.key === "Enter") {
+                setTrialFilterQuery(filterInput)
+              }
```

Comment:
[nit] This may cause issue for Japanese users. When using an IME, pressing Enter is used to confirm a candidate, which may unintentionally trigger the filter. 
<img width="725" height="334" alt="Screenshot 2025-08-04 18 13 08" src="https://github.com/user-attachments/assets/2be357d8-eecb-4038-b6d2-cda30ba6afe6" />